tests/test_filter_distribcell/test_filter_distribcell.py

```diff
@@ -0,0 +1,112 @@
+
+parser = OptionParser()
+parser.add_option('--mpi_exec', dest='mpi_exec', default='')
+parser.add_option('--mpi_np', dest='mpi_np', default='3')
+parser.add_option('--exe', dest='exe')
+(opts, args) = parser.parse_args()
+
+def test_run():
+
+    if opts.mpi_exec != '':
+        opts.mpi_exe = os.path.abspath(opts.mpi_exec)
+
+    opts.exe = os.path.abspath(opts.exe)
+
+    # Call 1
+    os.chdir('case-1')
+    cwd = os.getcwd()
+    if opts.mpi_exec != '':
+        proc = Popen([opts.mpi_exec, '-np', opts.mpi_np, opts.exe, cwd],
+               stderr=STDOUT, stdout=PIPE)
+    else:
+        proc = Popen([opts.exe, cwd], stderr=STDOUT, stdout=PIPE)
+    print(proc.communicate()[0])
+    returncode = proc.returncode
+    assert returncode == 0
+
+    # Call 2
+    os.chdir('../case-2')
+    cwd = os.getcwd()
+    if opts.mpi_exec != '':
+        proc = Popen([opts.mpi_exec, '-np', opts.mpi_np, opts.exe, cwd],
+               stderr=STDOUT, stdout=PIPE)
+    else:
+        proc = Popen([opts.exe, cwd], stderr=STDOUT, stdout=PIPE)
+    print(proc.communicate()[0])
+    returncode = proc.returncode
+    assert returncode == 0
+
+    # Call 3
+    os.chdir('../case-3')
+    cwd = os.getcwd()
+    if opts.mpi_exec != '':
+        proc = Popen([opts.mpi_exec, '-np', opts.mpi_np, opts.exe, cwd],
+               stderr=STDOUT, stdout=PIPE)
+    else:
+        proc = Popen([opts.exe, cwd], stderr=STDOUT, stdout=PIPE)
+    print(proc.communicate()[0])
+    returncode = proc.returncode
+    assert returncode == 0
+
+    os.chdir('..')
+
+def test_created_statepoint():
+    cwd = os.getcwd()
+    statepoint1 = glob.glob(cwd + '/case-1/statepoint.1.*')
+    statepoint2 = glob.glob(cwd + '/case-2/statepoint.1.*')
+    statepoint3 = glob.glob(cwd + '/case-3/statepoint.3.*')
+    assert len(statepoint1) == 1
+    assert len(statepoint2) == 1
+    assert len(statepoint3) == 1
+    string1 = statepoint1.pop()
+    string2 = statepoint2.pop()
+    string3 = statepoint3.pop()
+    assert string1.endswith('binary') or string1.endswith('h5')
+    assert string2.endswith('binary') or string2.endswith('h5')
+    assert string3.endswith('binary') or string3.endswith('h5')
+
+def test_results():
+    cwd = os.getcwd()
+    statepoint = list()
+    statepoint.append(glob.glob(cwd + '/case-1/statepoint.1.*'))
+    statepoint.append(glob.glob(cwd + '/case-2/statepoint.1.*'))
+    statepoint.append(glob.glob(cwd + '/case-3/statepoint.3.*'))
+    call(['python', 'results.py', statepoint.pop()[0], statepoint.pop()[0], statepoint.pop()[0]])
+    compare = filecmp.cmp('results_test.dat', 'results_true.dat')
+    if not compare:
+      os.rename('results_test.dat', 'results_error.dat')
+    assert compare
+
+def teardown():
+    cwd = os.getcwd()
+    output = glob.glob(cwd + '/statepoint.*')
+    output.append(cwd + '/geometry.xml')
+    output.append(cwd + '/materials.xml')
+    output.append(cwd + '/settings.xml')
+    output.append(cwd + '/tallies.xml')
```

Comment:
I'm not sure that the file deletion was necessary, I honestly don't recall
at this point. I'm tempted to say it had something to do with the first
test not being run correctly due to the old xml files being present, but if
that is not the case, then there is no need for deletion.

On Wed, May 27, 2015 at 1:09 PM, Will Boyd <notifications@github.com> wrote:

> In tests/test_filter_distribcell/test_filter_distribcell.py
> <https://github.com/mit-crpg/openmc/pull/379#discussion_r31156620>:
>
> > +    statepoint.append(glob.glob(cwd + '/case-1/statepoint.1.*'))
> > +    statepoint.append(glob.glob(cwd + '/case-2/statepoint.1.*'))
> > +    statepoint.append(glob.glob(cwd + '/case-3/statepoint.3.*'))
> > +    call(['python', 'results.py', statepoint.pop()[0], statepoint.pop()[0], statepoint.pop()[0]])
> > +    compare = filecmp.cmp('results_test.dat', 'results_true.dat')
> > +    if not compare:
> > +      os.rename('results_test.dat', 'results_error.dat')
> > +    assert compare
> > +
> > +def teardown():
> > +    cwd = os.getcwd()
> > +    output = glob.glob(cwd + '/statepoint.*')
> > +    output.append(cwd + '/geometry.xml')
> > +    output.append(cwd + '/materials.xml')
> > +    output.append(cwd + '/settings.xml')
> > +    output.append(cwd + '/tallies.xml')
>
> That is a very good question - I'm not sure why @laxderek
> <https://github.com/laxderek> implemented the test that way. I'll see if
> I can go ahead and remove this later today.
> … <#14d96596817abce9_>
> On May 27, 2015 5:28 AM, "Paul Romano" <notifications@github.com> wrote:
> In tests/test_filter_distribcell/test_filter_distribcell.py <
> https://github.com/mit-crpg/openmc/pull/379#discussion_r31127759>: > +
> statepoint.append(glob.glob(cwd + '/case-1/statepoint.1.*')) > +
> statepoint.append(glob.glob(cwd + '/case-2/statepoint.1.*')) > +
> statepoint.append(glob.glob(cwd + '/case-3/statepoint.3.*')) > +
> call(['python', 'results.py', statepoint.pop()[0], statepoint.pop()[0],
> statepoint.pop()[0]]) > + compare = filecmp.cmp('results_test.dat',
> 'results_true.dat') > + if not compare: > + os.rename('results_test.dat',
> 'results_error.dat') > + assert compare > + > +def teardown(): > + cwd =
> os.getcwd() > + output = glob.glob(cwd + '/statepoint.*') > +
> output.append(cwd + '/geometry.xml') > + output.append(cwd +
> '/materials.xml') > + output.append(cwd + '/settings.xml') > +
> output.append(cwd + '/tallies.xml') What's the purpose of deleting xml
> files during teardown? — Reply to this email directly or view it on GitHub <
> https://github.com/mit-crpg/openmc/pull/379/files#r31127759>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/379/files#r31156620>.
>